src/nvim/generators/gen_declarations.lua

```diff
@@ -225,14 +224,12 @@ while init ~= nil do
     else
       declline = declline - 1
     end
-  elseif init < declendpos then
-    -- Skipping over declaration
-  elseif is_needed_file then
+  elseif (not (init < declendpos)) and is_needed_file then
```

Comment:
We lost documentation here (the deleted comment). Would be better to add an inline `-- luacheck: ...` ignore.

Reply:
I have updated the code to reflect these changes.